docs/tools/config_system.md

```diff
@@ -1,23 +1,135 @@
+
+The Mbed OS build tools instruct the compiler to process the file `mbed_config.h` as if it were the first include of any C or C++ source file, so you do not have to include `mbed_config.h` manually.
+
+Do not edit `mbed_config.h` manually. It may be overwritten the next time you compile or export your application, and all your changes will be lost.
+
+### Configuration parameters in `mbed_app.json`, `mbed_lib.json`
+
+An application may have one `mbed_app.json` in the root of the application and many `mbed_lib.json` files throughout the application. When present, `mbed_app.json` may override library and target configuration parameters and define new configuration parameters. 
+
+#### Overriding configuration parameters
+
+The configuration system allows a user to override any defined configuration parameter with a JSON object named "target_overrides".
+
+The keys in the "target_overrides" section are the names of a target that the overrides apply to, or the special wildcard `*` that applies to all targets. The values within the "target_overrides" section are an object mapping configuration parameters, as printed by `mbed compile --conifg`, to new values. An example "target_overrides" section is provided below.
+
+```JSON
+"target_overrides": {
+    "*": {
+        "cellular.random_max_start_delay": "100"
+    },
+    "K64F": {
+        "cellular.use-apn-lookup": false
+    }
+}
+```
+
+Examining the configuration for the target `LPC1768` with `mbed compile --config -m LPC1768` results in the following configuration:
+
+```
+Configuration parameters
+------------------------
+cellular.random_max_start_delay = 100 (macro name: "MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY")
+cellular.use-apn-lookup = 1 (macro name: "MBED_CONF_CELLULAR_USE_APN_LOOKUP")
+<output truncated for brevity>
+```
+
+Examining the configuration for the target `K64F` with `mbed compile --config -m K64F` results in the following configuration:
+
+```
+Configuration parameters
+------------------------
+cellular.random_max_start_delay = 100 (macro name: "MBED_CONF_CELLULAR_RANDOM_MAX_START_DELAY")
+cellular.use-apn-lookup = 0 (macro name: "MBED_CONF_CELLULAR_USE_APN_LOOKUP")
+<output truncated for brevity>
+```
+
+The order in which overrides are considered is:
+
+ 1. Libraries override target configuration with `mbed_lib.json`.
+ 2. The application overrides target and library configuration with `mbed_app.json`
 
 #### Defining configuration parameters
 
-The configuration system understands configuration data defined in targets, libraries and applications. Although there are some slight differences in the way the configuration system works in these cases, the configuration parameters are always defined in a JSON object called "config".
+The configuration system understands configuration parameters defined in targets, libraries and applications. The configuration parameters are always defined in a JSON object called "config".
```

Comment:
"Target overrides" aren't considered "configuration parameters"?

Reply:
"target_overrides" override "configuration parameters". All keys within a value in "target_overrides" must name a "configuration parameter", so while they are not defining config, they're part of the same system. Where could I make this more clear? alternatively, what gave you the impression that "target overrides" are not part of the config system?

Reply:
Not sure what I was thinking here. Ignore, I guess.

Reply:
If you can remember what caused that confusion, pleas don't hesitate to let me know. I would like to improve the text to avoid it.